packages/client/hmi-client/src/components/workflow/scenario-templates/value-of-information/value-of-information-scenario.ts

```diff
@@ -0,0 +1,317 @@
+	}
+
+	setModelConfigSpec(id: string) {
+		this.modelConfigSpec.id = id;
+	}
+
+	setCalibrateSpec(ids: string[]) {
+		this.simulateSpec.ids = ids;
+	}
+
+	addInterventionSpec() {
+		this.interventionSpecs.push({ id: '' });
+	}
+
+	setInterventionSpec(id: string, index: number) {
+		this.interventionSpecs[index].id = id;
+	}
+
+	deleteInterventionSpec(index: number) {
+		this.interventionSpecs.splice(index, 1);
+	}
+
+	addParameter() {
+		this.parameters.push(null);
+	}
+
+	removeParameter(index: number) {
+		this.parameters.splice(index, 1);
+	}
+
+	setParameter(parameter: ParameterSemantic, index: number) {
+		formatUniformDistribution(parameter);
+		this.parameters[index] = parameter;
+	}
+
+	toJSON() {
+		return {
+			templateId: ValueOfInformationScenario.templateId,
+			workflowName: this.workflowName,
+			modelSpec: this.modelSpec,
+			modelConfigSpec: this.modelConfigSpec,
+			simulateSpec: this.simulateSpec
+		};
+	}
+
+	isValid(): boolean {
+		return (
+			!!this.workflowName &&
+			!!this.modelSpec.id &&
+			!!this.modelConfigSpec.id &&
+			!this.interventionSpecs.some((interventionSpec) => !interventionSpec.id) &&
+			!this.parameters.some((parameter) => !parameter) &&
+			!_.isEmpty(this.simulateSpec.ids)
+		);
+	}
+
+	async createWorkflow() {
+		const wf = new workflowService.WorkflowWrapper();
+		wf.setWorkflowName(this.workflowName);
+		wf.setWorkflowScenario(this.toJSON());
+
+		// 1. Add model and compare dataset nodes
+		const modelNode = wf.addNode(
+			ModelOp,
+			{ x: 0, y: 0 },
+			{
+				size: OperatorNodeSize.medium
+			}
+		);
+
+		wf.updateNode(modelNode, {
+			state: {
+				modelId: this.modelSpec.id
+			},
+			output: {
+				value: [this.modelSpec.id]
+			}
+		});
+
+		const compareDatasetNode = wf.addNode(
+			CompareDatasetOp,
+			{ x: 0, y: 0 },
+			{
+				size: OperatorNodeSize.medium
+			}
+		);
+
+		// add input ports for each simulation to the dataset transformer, this will be a matrix of intervention x parameter
+		for (let i = 0; i < this.interventionSpecs.length * this.parameters.length; i++) {
+			workflowService.appendInputPort(compareDatasetNode, {
+				type: 'datasetId|simulationId',
+				label: 'Dataset or Simulation'
+			});
+		}
+
+		let compareDatasetIndex = 0;
+
+		const modelConfig = await getModelConfigurationById(this.modelConfigSpec.id);
+
+		// chart settings for simulate node
+		let simulateChartSettings: ChartSetting[] = [];
+		simulateChartSettings = updateChartSettingsBySelectedVariables(
+			simulateChartSettings,
+			ChartSettingType.VARIABLE,
+			this.simulateSpec.ids
+		);
+
+		// 2. create model config nodes for each paramter and attach them to the model node
+		const modelConfigPromises = this.parameters.flatMap(async (parameter) => {
+			if (!parameter) return null;
+			const clonedModelConfig = _.cloneDeep(modelConfig);
+
+			const modelConfigNode = wf.addNode(
+				ModelConfigOp,
+				{ x: 0, y: 0 },
+				{
+					size: OperatorNodeSize.medium
+				}
+			);
```

Comment:
Food for thoughts, you have done several scenarios now - if you feel like these addNode/addEdge is a good pattern and also workable for the foreseeable future scenarios, you may want to consider cleaning things up a bit as there are a lot of repeated things (the xy-coodinate and the operator size for example) and have wrapper functions. I know you can definitely do it for addNode, not really sure about addEdge since it might be port-position dependent. 

Anyway food for thought in the future and not needed for this PR, I will leave this with you.

Reply:
Ya I have already made a ticket for this todo eventually: https://github.com/DARPA-ASKEM/terarium/issues/5699